cmd/utils/flags.go

```diff
@@ -1887,7 +1887,7 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 		cfg.DirectBroadcast = ctx.Bool(DirectBroadcastFlag.Name)
 	}
 	if ctx.IsSet(DisableSnapProtocolFlag.Name) {
-		cfg.DisableSnapProtocol = ctx.Bool(DisableSnapProtocolFlag.Name)
+		cfg.DisableSnapProtocol = cfg.DisableSnapProtocol || ctx.Bool(DisableSnapProtocolFlag.Name)
```

Comment:
does the flag not have the prority now?

Reply:
The snap protocol will be disabled as long as one of these 2 conditions is satisfied:

1. `tries-verify-mode` != `local`
2. `--disablesnapprotocol` enabled

On that note, the additional logic check `cfg.DisableSnapProtocol` is not needed because then the `DisableSnapProtocol` in the config will supersede the one specified by the flag. I got confused with the ordering, reverted the changes.